### PR TITLE
Chore: Start harmonizing linting with plugin SDK

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
Start harmonizing with the stricter linting scheme of the plugin SDK. Fix issues arising from enabling a couple more linters, a lot of which are just due to files not being formatted with `goimports`.

Also moving golangci-lint configuration files to [TOML](https://github.com/toml-lang/toml) format, since we use that for the SDK and it's a less troublesome (i.e., far less ambiguous) language than YAML.

### CODE REVIEW

**Review by ryantxu — APPROVED** [review]

**Review by jessabe — APPROVED** [review]

File: pkg/infra/usagestats/usage_stats.go [hunk 1]
```diff
@@ -181,9 +181,11 @@ func (uss *UsageStatsService) sendUsageStats(oauthProviders map[string]bool) {
 
 	client := http.Client{Timeout: 5 * time.Second}
 	go func() {
-		if _, err := client.Post(usageStatsURL, "application/json", data); err != nil {
+		resp, err := client.Post(usageStatsURL, "application/json", data)
+		if err != nil {
 			metricsLogger.Error("Failed to send usage stats", "err", err)
 		}
+		defer resp.Body.Close()
```
Comment (marefr) [inline pkg/infra/usagestats/usage_stats.go:188]: Defer needed here?

File: pkg/api/team_test.go [hunk 2]
```diff
@@ -9,11 +9,12 @@ import (
 	"github.com/grafana/grafana/pkg/components/simplejson"
 	"github.com/grafana/grafana/pkg/models"
 
+	"net/http"
```
Comment (marefr) [inline pkg/api/team_test.go:12]: Not sure what the recommendation/standard is, but I have something is VS code that sorts imports when saving a file creating something like this:
```go
// ordered set of builtin Go packages

// ordered set of external Go packages
```

File: pkg/models/datasource_cache_test.go [hunk 3]
```diff
@@ -224,13 +224,10 @@ func TestDataSourceProxyCache(t *testing.T) {
 			// 3. Send test request which should have the Authorization header set
 			req := httptest.NewRequest("GET", backend.URL+"/test-headers", nil)
 			res, err := transport.RoundTrip(req)
-			if err != nil {
-				log.Fatal(err.Error())
-			}
+			So(err, ShouldBeNil)
+			t.Cleanup(func() { res.Body.Close() })
```
Comment (marefr) [inline pkg/models/datasource_cache_test.go:228]: Nit, defer should do it here? Already using defer for `backend.Close()` above

**Review by marefr — CHANGES_REQUESTED** [review]
Great. Some minor comments/nits.

File: pkg/api/team_test.go (hunk 2 above)
Comment (aknuds1) [reply pkg/api/team_test.go:12]: @marefr I don't have any control of it, I think it's what the `goimports` formatter does automatically.

File: pkg/infra/usagestats/usage_stats.go (hunk 1 above)
Comment (aknuds1) [reply pkg/infra/usagestats/usage_stats.go:188]: Good catch, I also noticed you shouldn't try to close here if an error was returned.

**Comment by kylebrandt** [comment]
I think `nakedret`'s default value of 5 is too aggressive:

"nakedret is a Go static analysis tool to find naked returns in functions greater than a specified function length." ... " If not specified, it defaults to 5."

That being said, I'm not to fond of this at all. I don't think it is going to make a big difference in code quality (named params in returns are rare anyways) and more likely just to be a annoying this randomly run into with PRs.

**Comment by aknuds1** [comment]
What do you suggest regarding nakedret @kylebrandt? Disabling it? I have too little experience with it to say. It caught very little in Grafana at least, so not very annoying so far.

**Comment by kylebrandt** [comment]
> What do you suggest regarding nakedret @kylebrandt? Disabling it? I have too little experience with it to say. It caught very little in Grafana at least, so not very annoying so far.

Yes disabling it. It catching little is kind of my point - that it will happen infrequently enough that it might feel like "random build failures", but not frequently enough that it is the sort of thing people will remember, or frequently enough to really change the code quality in a significant way.

Or to put it another way - I think that linter it is too nitpicky for this codebase :-)

**Comment by aknuds1** [comment]
@kylebrandt OK I can disable nakedret.

**Review by kylebrandt — APPROVED** [review]

**Comment by aknuds1** [comment]
@marefr Please have another look.

**Review by bergquist — APPROVED** [review]
LGTM

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**